crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -293,25 +241,25 @@ where
             // TODO: Consider implementing optional re-sync of the piece instead of just forgetting
             WorkerCommand::ForgetKey { key } => {
                 let mut caches = self.piece_caches.write().await;
-                let Some(offset) = caches.stored_pieces.remove(&key) else {
+                let Some(offset) = caches.remove_stored_piece(&key) else {
                     // Key not exist
                     return;
                 };
-                let cache_index = usize::from(offset.cache_index);
```

Comment:
Why removing the variable here and below? It just increases the diff and makes logging more cluttered. If you no longer need `usize::from()` call then make it `let cache_index = offset.cache_index;` and keep the rest the same as it was.

Reply:
`usize::from()` is only used when get_backend, so I moved this action inside the method.
I forced this in `chore: move PieceCacheState into submodule` commit(only changed this)